frontend/catalyst/qjit_device.py

```diff
@@ -218,3 +220,112 @@ def _decomp_to_unitary(self, *_args, **_kwargs):
 
         self.check_validity(expanded_tape.operations, [])
         return expanded_tape
+
+
+class QJITDeviceNewAPI(qml.devices.Device):
+    """QJIT device for the new device API.
+    A device that interfaces the compilation pipeline of Pennylane programs.
+    Args:
+        wires (Shots): the number of wires to initialize the device with
+        shots (int): How many times the circuit should be evaluated (or sampled) to estimate
+            the expectation values. Defaults to ``None`` if not specified. Setting
+            to ``None`` results in computing statistics like expectation values and
+            variances analytically
+        backend_name (str): name of the device from the list of supported and compiled backend
+            devices by the runtime
+        backend_kwargs (Dict(str, AnyType)): An optional dictionary of the device specifications
+    """
+
+    # These must be present even if empty.
+    operations = []
+    observables = []
+
+    operations_supported_by_QIR_runtime = RUNTIME_OPERATIONS
+
+    @staticmethod
+    def _check_mid_circuit_measurement(config):
+        return config["compilation"]["mid_circuit_measurement"]
```

Comment:
Suggestion: I don't know if all these methods are the same or not, but if they were one could consider a common base class for QJITDevice and QJITDeviceNewAPI.

Reply:
I have to check the new status with Sergei's PR, we could have double inheritance (one from the PennyLane device and one from a base class).